src/Swift.Bindings/src/Model/TypeDecl/BaseDecl.cs

```diff
@@ -11,7 +11,7 @@ public record BaseDecl
         /// <summary>
         /// Name of the declaration.
         /// </summary>
-        public required string Name { get; set; }
+        public required string Name { get; set; } //TODO: Hide or remove this property. This might not contain a correct name.
```

Comment:
Could you provide more context to this comment?

Reply:
Each node in `abi.json` has a name field. However this name field is not necessary the thing we expect. For example for generic arguments name is always `GenericTypeParam`. As there is no direct mapping between `node.Name` and `BaseDecl.Name` I think this might get a bit confusing long run.

Also, different decls will have different requirements about the name. For example for types we might need to get to the fully qualified path, the fully qualified path without module or just type name depending on context. We don't need this for e.g. `ArgumentDecl`.

I think we should drop `BaseDecl.Name` and replace it with more specialised name types on the relevant `Decls`.